iothub/device/src/Pipeline/RetryDelegatingHandler.cs

```diff
@@ -388,53 +484,68 @@ public override async Task OpenAsync(CancellationToken cancellationToken)
             // The current behavior does not support open->close->open
             if (_isDisposed)
             {
-                throw new ObjectDisposedException(nameof(RetryDelegatingHandler));
+                throw new ObjectDisposedException("IoT client", ClientDisposedMessage);
             }
-
-            if (_isOpen)
+            switch (GetClientTransportStatus())
             {
-                return;
-            }
-
-            await _handlerSemaphore.WaitAsync(cancellationToken).ConfigureAwait(false);
-            try
-            {
-                if (!_isOpen)
-                {
-                    if (Logging.IsEnabled)
-                        Logging.Info(this, "Opening connection", nameof(OpenAsync));
+                case ClientTransportStatus.Open:
+                case ClientTransportStatus.Opening:
+                    return;
```

Comment:
If the client is currently open or opening, simply exit.

Reply:
Updating the formatting marked this comment as outdated. Resolving this and adding a new one.